tests/deploy_test.sh

```diff
@@ -0,0 +1,600 @@
+#!/bin/bash
+
+include './src/deploy.sh'
+include './tests/utils.sh'
+
+FAKE_KERNEL="tests/.tmp"
+
+# Some of the functions invoked by kw need to be mocked; otherwise, we cannot
+# test all the elements in the function. The following functions try to mimic
+# some of these functions behaviour.
+function which_distro_mock()
+{
+  echo "debian"
+}
+
+function get_kernel_release_mock()
+{
+  echo '5.4.0-rc7-test'
+}
+
+function get_kernel_version_mock()
+{
+  echo '5.4.0-rc7'
+}
+
+function root_id_mock()
+{
+  echo "0"
+}
+
+function setUp()
+{
+  local create_mkinitcpio="$1"
+
+  rm -rf "$FAKE_KERNEL"
+
+  # This creates tests/.tmp which should mock a kernel tree root. A .git
+  # dir is also created inside tests/.tmp so that get_maintainer.pl thinks
+  # it is a git repo. This is done in order to avoid some warnings that
+  # get_maintainer.pl prints when no .git is found.
+  mk_fake_kernel_root "$FAKE_KERNEL"
+
+  cp -f "$MAINTAINERS_SAMPLE" "$FAKE_KERNEL"
+  cp -f "$CHECKPATH_EXT" "$FAKE_KERNEL/scripts/"
+
+  export preset_name="template_mkinitcpio.preset"
+  export test_path="$PWD/$FAKE_KERNEL"
+  export KW_PLUGINS_DIR="$test_path"
+  export KW_CACHE_DIR="$test_path"
+  export KW_ETC_DIR="$PWD/$SAMPLES_DIR/etc"
+  export DEPLOY_SCRIPT="$test_path/$kernel_install_path/deploy.sh"
+  export KW_PLUGINS_DIR="$PWD/src/plugins/"
```

Comment:
Maybe have just one `KW_PLUGINS_DIR` export? (see line 48)

Reply:
nice catch, I guess we can remove the first export